nvchecker: 2.18 -> 2.19; modernize

mdaniels5757 opened the pull request:
Upstream changes: https://github.com/lilydjwg/nvchecker/compare/176b08fc665c759b84832fab771ccc588fd81915..e513efb6adf8471effe8c822a33098684c3bca38

Closes: #444339
Closes: #447405

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [X] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mdaniels5757 committed:
python3Packages.nvchecker: 2.18 -> 2.19

Co-authored-by: R. Ryantm <ryantm-bot@ryantm.com>

mdaniels5757 committed:
python313Packages.nvchecker: modernize

- adjust ordering

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 444446`
Commit: [`01f6ea84ce00935f936ad2cf620a1a47ed46a364`](https://github.com/NixOS/nixpkgs/commit/01f6ea84ce00935f936ad2cf620a1a47ed46a364) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/01f6ea84ce00935f936ad2cf620a1a47ed46a364..pull/444446/head))
Merge: [`5db99556fa40e5a4f155bc461ef394e0be5a1c00`](https://github.com/NixOS/nixpkgs/commit/5db99556fa40e5a4f155bc461ef394e0be5a1c00)

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/17870154604


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.nvfetcher.doc</li>
    <li>nvchecker</li>
    <li>nvchecker.dist</li>
    <li>nvfetcher</li>
    <li>python312Packages.nvchecker</li>
    <li>python312Packages.nvchecker.dist</li>
    <li>python313Packages.nvchecker</li>
    <li>python313Packages.nvchecker.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.nvfetcher.doc</li>
    <li>nvchecker</li>
    <li>nvchecker.dist</li>
    <li>nvfetcher</li>
    <li>python312Packages.nvchecker</li>
    <li>python312Packages.nvchecker.dist</li>
    <li>python313Packages.nvchecker</li>
    <li>python313Packages.nvchecker.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.nvfetcher.doc</li>
    <li>nvchecker</li>
    <li>nvchecker.dist</li>
    <li>nvfetcher</li>
    <li>python312Packages.nvchecker</li>
    <li>python312Packages.nvchecker.dist</li>
    <li>python313Packages.nvchecker</li>
    <li>python313Packages.nvchecker.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.nvfetcher.doc</li>
    <li>nvchecker</li>
    <li>nvchecker.dist</li>
    <li>nvfetcher</li>
    <li>python312Packages.nvchecker</li>
    <li>python312Packages.nvchecker.dist</li>
    <li>python313Packages.nvchecker</li>
    <li>python313Packages.nvchecker.dist</li>
  </ul>
</details>

Sigmanificient commented:
Am i doing something wrong? I used the example from the doc:
```
[nix-shell:~/.cache/nixpkgs-review/pr-444446]$ nvfetcher -c nvfetcher.toml 
nvfetcher: Failed to parse config file: Decode error at '.src': Field does not exist

CallStack (from HasCallStack):
  error, called at app/Main.hs:33:15 in main:Main
```

mdaniels5757 commented:
Hmm, `nvfetcher` works for me...

```
> gh co 444446
...
> nix-build -A nvfetcher
...
> cat nvfetcher.toml
[rustdeck]
src.github = "rustdesk/rustdesk"
fetch.github = "rustdesk/rustdesk"
# Compute outputHashes for git dependencies in cargo lock
cargo_lock = ["Cargo.lock"]
> ./result/bin/nvfetcher
# CheckGitHubRelease
    owner: rustdesk
    repo: rustdesk
# FetchGitHub
  owner: rustdesk
  repo: rustdesk
  rev: 1.4.2
  deepClone: False
  fetchSubmodules: False
  leaveDotGit: False
  sparseCheckout: []
Changes:
rustdeck: ∅ → 1.4.2
```

mdaniels5757 commented:
@Sigmanificient Could you please take another look at this if you have the chance?

mdaniels5757 commented:
@Sigmanificient Ah. It looks like you're taking an `nvchecker` config file, and passing it to `nvfetcher`?

`nvfetcher` depends on `nvchecker`, but they use completely different config file formats. And when I run nvfetcher on an nvchecker config file, I get the same error you do:

```
> nix-shell -p nvchecker nvfetcher
> cat nvchecker.toml # This is valid for nvchecker, but not nvfetcher
[nvchecker]
source = "github"
github = "lilydjwg/nvchecker"

[python-toml]
source = "pypi"
pypi = "toml"

> cat nvfetcher.toml # Valid for nvfetcher, but not nvchecker
[rustdeck]
src.github = "rustdesk/rustdesk"
fetch.github = "rustdesk/rustdesk"
# Compute outputHashes for git dependencies in cargo lock
cargo_lock = ["Cargo.lock"]

> nvchecker -c nvchecker.toml # Correct config file for this program; works
[I 09-30 17:13:23.995 core:416] python-toml: updated to 0.10.2 url=https://pypi.org/project/toml/0.10.2/
[I 09-30 17:13:24.258 core:416] nvchecker: updated to 20250927.101842 revision=e9d181ee13236f734fb74f1e9890b2e5e08aae54 url=https://github.com/lilydjwg/nvchecker/commit/e9d181ee13236f734fb74f1e9890b2e5e08aae54

> nvfetcher -c nvchecker.toml # Incorrect config file for this program, seemingly what you were doing?
nvfetcher: Failed to parse config file: Decode error at '.src': Field does not exist

CallStack (from HasCallStack):
  error, called at app/Main.hs:33:15 in main:Main

> nvchecker -c nvfetcher.toml # Incorrect config file for this program
[E 09-30 17:19:16.896 core:366] rustdeck: no source specified
[E 09-30 17:19:16.896 core:448] rustdeck: no-result error="GetVersionError('no source specified')"

> nvfetcher -c nvfetcher.toml  # Correct config file for this program; works
# CheckGitHubRelease
    owner: rustdesk
    repo: rustdesk
Changes:
rustdeck: ∅ → 1.4.2
```

Sigmanificient reviewed (approved):
Yep, i am a mess 😅

mdaniels5757 commented:
> Yep, i am a mess 😅

@Sigmanificient Lol you're fine. No worries at all, and thanks for the merge!
